openlcsd/flow/tasks.py

```diff
@@ -78,6 +79,40 @@ def get_config(context, engine):
     engine.logger.info("Worker {}".format(socket.gethostname()))
 
 
+def filter_duplicate_import(context, engine):
+    url = 'check_duplicate_import'
+    cli = context.get('client')
+    component_type = context.get('component_type')
+    build_data = context.get('build', {})
+    data = {
+        'name': build_data.get('name', ''),
+        'version': build_data.get('version', ''),
+        'release': build_data.get('release', ''),
+        'type': component_type,
+        'license_scan': context.get('license_scan'),
+        'copyright_scan': context.get('copyright_scan')
+    }
+    if component_type == 'RPMMOD':
+        data['arch'] = ''
+    else:
+        data['arch'] = build_data.get('arch', 'src')
+    msg = 'Start to check duplicate import for {}'.format(data)
+    engine.logger.info(msg)
+    resp = cli.post(url, data=data)
+    if resp.status_code == 200:
+        results = resp.json()['results']
+        if results:
+            obj_url = results.get('obj_url')
+            if obj_url:
+                task_id = context.get('task_id')
+                msg = 'Found duplicate import for task: {}, \
+                        obj_url: {}'.format(task_id, obj_url)
+                engine.logger.warning(msg)
+                res = AsyncResult(task_id)
+                res.revoke(terminate=True)
```

Comment:
Per feedback, task will be marked as duplicate instead of revoking here.

Reply:
DONE